cli/src/main.rs

```diff
@@ -256,6 +267,14 @@ fn main() -> Result<(), Box<dyn error::Error>> {
             .global(true)
             .help(SKIP_SEED_PHRASE_VALIDATION_ARG.help),
     )
+    .arg(
+        Arg::with_name("rpc_timeout")
+            .long("rpc-timeout")
+            .value_name("SECONDS")
+            .takes_value(true)
+            .global(true)
+            .help("Timeout value for RPC requests"),
```

Comment:
Should we use a `validator` to impose some sort of reasonable max?

Reply:
An upper bound seems a little dubious.  I mean 30s _should_ be plenty, but here we are :sweat_smile: 

What are you thinking, something like 30-60min?

Reply:
I guess I was imagining something like 10min.
But I'll defer to your wisdom about the api nodes, if you don't think it will be a problem to have no limit.

Reply:
The servers really shouldn't be leaving that decision up to their clients.  I'm not sure how big the lift would be to add server-side request timeouts though.

I'd also be OK with hiding this arg and not promoting it be used widely

Reply:
And hidden!

https://github.com/solana-labs/solana/pull/11181/commits/d3c622b362f85637877dfcb4529c3fbe0c695190#diff-4c438f60247f494e2aee67fcce9e11e0R274